hw/xbox/xid-sdl.c

```diff
@@ -488,7 +488,7 @@ static void usb_xbox_gamepad_realize(USBDevice *dev, Error **errp)
     s->out_state_capabilities.report_id = 0;
 
     /* FIXME: Make sure SDL was init before */
-    if (SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER)) {
+    if (SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER | SDL_INIT_HAPTIC)) {
```

Comment:
The haptic subsystem might be optional (we'll have to check the SDL docs), so we should probably initialize that separately to get meaningful error messages + disable haptic support if necessary.

Before changing anything, consult the SDL docs.

Reply:
Without `SDL_INIT_HAPTIC`, `SDL_HapticOpen` always returns -1 and `SDL_GetError` reports that the device does not support haptic feedback, regardless of feedback support. Without `SDL_HapticOpen` I can't even sniff for haptic support with `SDL_HapticQuery`, so I think it is required here.

Docs are fairly thin on this matter: https://wiki.libsdl.org/CategoryForceFeedback

Reply:
To be clear, I am for a separate init to handle the error gracefully though.

Reply:
Right - you still have to init it obviously. But it should probably be done like this:

```c
if (SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER)) { /* handle errors for gamecontroller [abort?] */ }
if (SDL_InitSubSystem(SDL_INIT_HAPTIC)) { /* handle errors for haptic [disable / skip force-feedback] */ }
```

Because only that way we can cleanly differentiate between errors of the *required* gamecontroller subsystem, and the *optional* haptic subsystem.

*(Or whatever SDL recommens / what other projects do that ran into similar issues)*

Reply:
Makes sense, I miss-understood what you meant by optional. I think we are both in agreement here. Good idea.